src/pages/main/components/SortPanel/SortPanel.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useCallback, useState } from 'react';
+import classes from './SortPanel.module.scss';
+
+interface SortPanelProps {
+  onChangeSortingDirection?: (isDownDirection: boolean) => void;
+}
+
+const SortPanel: React.FC<SortPanelProps> = ({ onChangeSortingDirection = () => {} }) => {
+  const [isDown, changeDirection] = useState(true);
+
+  const onChangeDirection = useCallback((isDown: boolean) => {
+    changeDirection(!isDown);
+    onChangeSortingDirection(!isDown);
+  }, []);
```

Comment:
because usage of arrow function inside component `onClick` is not a best practice, a possible way to use a callback. But your way is ok.
```
const onChangeDirection = useCallback(() => {
    onChangeSortingDirection(!isDown);
  }, [isDown]);

<Component onClick={onChangeDirection}>

```